### PR TITLE
Update pre-commit and constraints and minor fixes for 0.4.19 release

### PR DESCRIPTION
**Pull request by Czaki**
# Description

update meta-tool for 0.4.19 release

### CODE REVIEW

**Comment by Czaki** [comment]
@napari-bot update constraints

**Comment by Czaki** [comment]
@napari-bot update constraints

**Review by jni — APPROVED** [review]

**Comment by Czaki** [comment]
running test locally on windows ends with success...

File: napari/layers/image/image.py [hunk 1]
```diff
@@ -917,7 +917,12 @@ def _update_thumbnail(self):
             image = np.max(image, axis=0)
 
         # float16 not supported by ndi.zoom
-        dtype = np.dtype(image.dtype)
+        try:
+            dtype = np.dtype(image.dtype)
+        except TypeError:
+            # tensorstore case
+            dtype = np.dtype(image.dtype.type)
+
```
Comment (Carreau) [inline napari/layers/image/image.py:925]: Can we be more explicit here and have an if isinstance(...) or something else ?

**Review by Carreau — APPROVED** [review]
One minor comment on a try/except

File: napari/layers/image/image.py (hunk 1 above)
Comment (Czaki) [reply napari/layers/image/image.py:925]: I do not know how. Problem is with tensorstore dtype, and tensorstore is optional depenecy.

Comment (Carreau) [reply napari/layers/image/image.py:925]: Hum, can we create an issue once this is merge to track being explicit ? 
Something like `image.dtype.__class__.__name__.startswith('tensorstore.')` ?

Comment (Czaki) [reply napari/layers/image/image.py:925]: I do not observe this problem on the main branch. It happens only on the async test on 0.4.19 branch.

Comment (Carreau) [reply napari/layers/image/image.py:925]: Oh, this target 4.19 ! Sorry, then nevermind, this is all good.

**Comment by Czaki** [comment]
@napari-bot update constraints

**Pull request merged**